langchain/src/stores/message/momento.ts

```diff
@@ -0,0 +1,180 @@
+   * If not specified, the cache client default is used.
+   * @param {boolean} props.ensureCacheExists If true, ensure that the cache exists before returning.
+   * If false, the cache is not checked for existence.
+   * @throws {InvalidArgumentError} If {@link props.sessionTtl} is not strictly positive.
+   * @returns A new chat message history backed by Momento.
+   */
+  public static async CreateAsync(
+    props: MomentoChatMessageHistoryProps
+  ): Promise<MomentoChatMessageHistory> {
+    const instance = new MomentoChatMessageHistory(props);
+    if (props.ensureCacheExists || props.ensureCacheExists === undefined) {
+      await ensureCacheExists(props.client, props.cacheName);
+    }
+    return instance;
+  }
+
+  /**
+   * Validate the user-specified TTL, if provided, is strictly positive.
+   * @param ttlSeconds The TTL to validate.
+   */
+  private validateTtlSeconds(ttlSeconds?: number): void {
+    if (ttlSeconds !== undefined && ttlSeconds <= 0) {
+      throw new InvalidArgumentError("ttlSeconds must be positive.");
+    }
+  }
+
+  public async getMessages(): Promise<BaseChatMessage[]> {
+    const fetchResponse = await this.client.listFetch(
+      this.cacheName,
+      this.sessionId
+    );
+
+    let messages: StoredMessage[] = [];
+    if (fetchResponse instanceof CacheListFetch.Hit) {
+      messages = fetchResponse
+        .valueList()
+        .map((serializedStoredMessage) => JSON.parse(serializedStoredMessage));
+    } else if (fetchResponse instanceof CacheListFetch.Miss) {
+      // pass
+    } else if (fetchResponse instanceof CacheListFetch.Error) {
+      throw fetchResponse.innerException();
+    } else {
+      throw new Error(`Unknown response type: ${fetchResponse.toString()}`);
+    }
+    return mapStoredMessagesToChatMessages(messages);
+  }
+
+  public async addMessage(message: BaseChatMessage): Promise<void> {
+    const messageToAdd = JSON.stringify(
+      mapChatMessagesToStoredMessages([message])[0]
+    );
+
+    const pushResponse = await this.client.listPushBack(
+      this.cacheName,
+      this.sessionId,
+      messageToAdd,
+      { ttl: this.sessionTtl }
+    );
+    if (pushResponse instanceof CacheListPushBack.Success) {
+      // pass
+    } else if (pushResponse instanceof CacheListPushBack.Error) {
+      throw pushResponse.innerException();
+    } else {
+      throw new Error(`Unknown response type: ${pushResponse.toString()}`);
+    }
+  }
+
+  public async clear(): Promise<void> {
+    const deleteResponse = await this.client.delete(
+      this.cacheName,
+      this.sessionId
+    );
+    if (deleteResponse instanceof CacheDelete.Success) {
+      // pass
+    } else if (deleteResponse instanceof CacheDelete.Error) {
+      throw deleteResponse.innerException();
+    } else {
+      throw new Error(`Unknown response type: ${deleteResponse.toString()}`);
+    }
+  }
+}
+
+/**
+ * Ensure that the cache exists.
+ */
+async function ensureCacheExists(
```

Comment:
Should this not be on the class?

Reply:
Good catch. Since this code is common to both the cache and chat history modules, I intended to extract to a common utility module. I've made this change in https://github.com/hwchase17/langchainjs/pull/1416/commits/b0750e9c1ce3171225d89b5eb8d5a0ccdae620cb.